### PR TITLE
Mbs 9655 extract test summary logic from runner

### PR DESCRIPTION
**Pull request by dsvoronin**
- [x] flakyReport should be in separate step (need to enable it for PR's)

### CODE REVIEW

**Comment by dsvoronin** [comment]
flakyReports should be in separate ci-step

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/InMemoryReport.kt [hunk 1]
```diff
@@ -14,6 +15,9 @@ class InMemoryReport(private val id: String) : Report {
     private val testStatusFinalizer = TestStatusFinalizer.create()
     private val testAttempts = mutableListOf<AndroidTest>()
 
+    override val reportApi: ReportsApi
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/InMemoryReport.kt]: I consider it mustn't know about reportApi

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/Report.kt [hunk 2]
```diff
@@ -17,6 +17,8 @@ import java.io.Serializable
 
 interface Report : ReadReport {
 
+    val reportApi: ReportsApi
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/Report.kt]: Where could I look to realize why we need report API there?

File: subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/FlakyReportStep.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    override fun registerTask(project: Project, rootTask: TaskProvider<out Task>) {
+        require(project.pluginManager.hasPlugin(testSummaryPluginId)) {
+            "FlakyReport step can't be initialized without $testSummaryPluginId plugin applied"
+        }
+
+        require(configuration.isNotBlank()) {
+            "FlakyReport step can't be initialized without provided instrumentation configuration"
+        }
+
+        if (useImpactAnalysis && !project.internalModule.isModified()) return
```
Comment (eugene-krivobokov) [inline subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/FlakyReportStep.kt:27]: I'd prefer to skip the task but have it in the graph.

Comment (eugene-krivobokov) [reply subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/FlakyReportStep.kt:27]: Also, `onlyIf` is executed in an execution phase. We shouldn't avoid using heavy impact analysis in a configuration.

File: subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/Report.kt (hunk 2 above)
Comment (dsvoronin) [reply subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/report/Report.kt]: let me try to remove this (was a refactoring leftover)

File: subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/FlakyReportStep.kt (hunk 3 above)
Comment (dsvoronin) [reply subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/FlakyReportStep.kt:27]: well, i don't like this way to handle impact analysis as well; but seeing at as out of this pr scope change
It's how we deal with impact analysis in all steps

Comment (eugene-krivobokov) [reply subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/FlakyReportStep.kt:27]: Agree

File: subprojects/gradle/test-summary/src/main/kotlin/com/avito/test/summary/TestSummarySender.kt [hunk 4]
```diff
@@ -106,20 +117,27 @@ class TestSummarySenderImplementation(
...
+                        author = slackUserName,
                         emoji = slackEmojiProvider.emojiName(suite.percentSuccessOfAutomated)
                     )
                 )
             }.onFailure { throwable ->
-                logger.critical(
-                    "Can't compose slack message for summary: buildUrl=$buildUrl",
-                    throwable
-                )
+                logger.critical("Can't compose slack message for summary: buildUrl=$buildUrl", throwable)
```
Comment (sergioboi) [inline subprojects/gradle/test-summary/src/main/kotlin/com/avito/test/summary/TestSummarySender.kt]: warning

File: subprojects/gradle/test-summary/src/main/kotlin/com/avito/test/summary/TestSummarySender.kt [hunk 5]
```diff
@@ -106,20 +117,27 @@ class TestSummarySenderImplementation(
...
-}
-
-val summaryChannel = SlackChannel("#android-test-summary")
 
-private const val SLACK_USER_NAME = "Test Analyzer"
+    private fun ReportsApi.tryGetId(reportCoordinates: ReportCoordinates): String? {
+        return when (val result = getReport(reportCoordinates)) {
+            is GetReportResult.Found -> result.report.id
+            GetReportResult.NotFound -> {
+                logger.critical("Can't find report for runId=${reportCoordinates.runId}")
```
Comment (sergioboi) [inline subprojects/gradle/test-summary/src/main/kotlin/com/avito/test/summary/TestSummarySender.kt]: warning

File: subprojects/gradle/test-summary/src/main/kotlin/com/avito/test/summary/TestSummarySender.kt [hunk 6]
```diff
@@ -106,20 +117,27 @@ class TestSummarySenderImplementation(
...
+                null
+            }
+            is GetReportResult.Error -> {
+                logger.critical("Can't find report for runId=${reportCoordinates.runId}", result.exception)
```
Comment (sergioboi) [inline subprojects/gradle/test-summary/src/main/kotlin/com/avito/test/summary/TestSummarySender.kt]: warning

**Review by eugene-krivobokov — APPROVED** [review]

**Pull request merged**